mesh/pmesh.cpp

```diff
@@ -1292,6 +1292,66 @@ void ParMesh::Finalize(bool refine, bool fix_orientation)
    FinalizeParTopo();
 }
 
+void ParMesh::DistributeAttributes(Array<int> &attr)
+{
+   // Determine the largest attribute number across all processors
+   int max_attr = attr.Max();
+   int glb_max_attr = -1;
+   MPI_Allreduce(&max_attr, &glb_max_attr, 1, MPI_INT, MPI_MAX, MyComm);
+
+   // Create marker arrays to indicate which attributes are present
+   // assuming attribute numbers are in the range [1,glb_max_attr].
+   bool * attr_marker = new bool[glb_max_attr];
+   bool * glb_attr_marker = new bool[glb_max_attr];
```

Comment:
Overall, I feel like this this should work. I still need to run the tests for it. However, I am a little worried about the case where a user sets their attributes to some large number like the maximum allowable integer. Then we would be passing around a lot of data during the next ```MPI_Allreduce```. It might make sense to also get the minimum attribute number and then have the array bounded by the difference in the two. Although, we would still run into this issue of creating a large array if the user also set a value to like 1.

Reply:
Based on yesterday's talks, it sounds like this issue of large attributes numbers living on the mesh class would be best addressed in a future PR.